numpy/core/tests/test_multiarray.py

```diff
@@ -2864,10 +2864,10 @@ def test_swapaxes(self):
         assert_(a.flags['OWNDATA'])
         b = a.copy()
         # check exceptions
-        assert_raises(ValueError, a.swapaxes, -5, 0)
-        assert_raises(ValueError, a.swapaxes, 4, 0)
-        assert_raises(ValueError, a.swapaxes, 0, -5)
-        assert_raises(ValueError, a.swapaxes, 0, 4)
+        assert_raises(np.AxisError, a.swapaxes, -5, 0)
+        assert_raises(np.AxisError, a.swapaxes, 4, 0)
+        assert_raises(np.AxisError, a.swapaxes, 0, -5)
+        assert_raises(np.AxisError, a.swapaxes, 0, 4)
```

Comment:
These were found by temporarily removing the base classes from `AxisError`. They don't actually represent a compatibility problem, unless someone is doing something bizarre like `type(exc) is ValueError`.

Reply:
Agreed, this looks good to me.